apps/astarte_housekeeping/lib/astarte_housekeeping/engine.ex

```diff
@@ -229,11 +293,24 @@ defmodule Astarte.Housekeeping.Engine do
     merge_realm_status(old_realm, attrs)
   end
 
+  defp merge_realm_status(old_realm, %{datastream_maximum_storage_retention: nil} = update_attrs) do
+    attrs = Map.delete(update_attrs, :datastream_maximum_storage_retention)
+    merge_realm_status(old_realm, attrs)
+  end
+
   defp merge_realm_status(old_realm, %{device_registration_limit: :remove_limit} = update_attrs) do
     attrs = Map.delete(update_attrs, :device_registration_limit)
     merge_realm_status(%{old_realm | device_registration_limit: nil}, attrs)
   end
 
+  defp merge_realm_status(
+         old_realm,
+         %{datastream_maximum_storage_retention: 0} = update_attrs
+       ) do
+    attrs = Map.delete(update_attrs, :datastream_maximum_storage_retention)
+    merge_realm_status(%{old_realm | datastream_maximum_storage_retention: nil}, attrs)
+  end
+
```

Comment:
Put the two clauses dealing with `datastream_maximum_storage_retention` close so related stuff are close together. Also a comment explaining the difference between the two cases wouldn't hurt since it's not clear.